src/arch/xtensa/CMakeLists.txt

```diff
@@ -18,6 +18,8 @@ elseif(CONFIG_MT8188)
         set(platform_folder mt8188)
 elseif(CONFIG_MT8195)
 	set(platform_folder mt8195)
+else()
+	message(FATAL_ERROR "Platform not defined, check your Kconfiguration?")
```

Comment:
FWIW: you could move this to a build-time #error in a C file somewhere to avoid having to do this in two different cmake schemes.

Reply:
I respectfully but strongly disagree. I think it's better to check and fail as early as possible. Also, C compilation is parallelized and typically unreadable before adding `-j1` whereas CMake configuration is single-threaded. I think it's also better to always have a "default" error clause in a case/switch, and/or to check a variable is not empty before using it (the `-I platform/${EMPTY}/include` wasted a large amount of my time!)

I avoid CMake as much as possible but the duplication here is only one line. It's only because Zephyr versus non-Zephyr so it won't grow larger.

Finally, in which C file?

All this being said, if there is some C file where this check could be useful TOO then why not; "defence in depth".